clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy

```diff
@@ -296,13 +296,14 @@ class LoadBalancerV2UpsertHandler {
 
     // Get the state of this load balancer from aws
     List<TargetGroup> existingTargetGroups = []
-    if (targetGroups.size() > 0) {
-      try {
-        existingTargetGroups = loadBalancing.describeTargetGroups(new DescribeTargetGroupsRequest().withNames(targetGroups.collect { it.name }))?.targetGroups
-      } catch (TargetGroupNotFoundException ignore) {
-        // If a subset of the target groups requested does not exist, actually returns the target groups that do exist
-        // If none of the target groups requested exist, throws an exception instead.
-      }
+    try {
+      existingTargetGroups = loadBalancing.describeTargetGroups(
+        new DescribeTargetGroupsRequest().withLoadBalancerArn(loadBalancer.loadBalancerArn)
+      )?.targetGroups
+    } catch (TargetGroupNotFoundException ignore) {
```

Comment:
you're doing a different query now ... asking for `targetGroups` in a single `loadBalancerArn` ... are these exceptions still relevant?

If you're asking for one `loadBalancerArn` and you get a LoadBalancerNotFoundException, is that legitimate?

Reply:
Nope and Neeeeope.